src/server/zset_family.cc

```diff
@@ -2030,6 +2030,58 @@ void ZSetFamily::ZInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(smvec.size());
 }
 
+void ZSetFamily::ZInter(CmdArgList args, ConnectionContext* cntx) {
+  OpResult<SetOpArgs> op_args_res = ParseSetOpArgs(args, false);
+
+  if (!op_args_res) {
+    switch (op_args_res.status()) {
+      case OpStatus::INVALID_FLOAT:
+        return (*cntx)->SendError("weight value is not a float", kSyntaxErrType);
+      default:
+        return (*cntx)->SendError(op_args_res.status());
+    }
```

Comment:
This is the 3rd time this piece of code is pasted in this file, maybe we could extract it to some function?
(please put that function in an anonymous namespace, i.e. `namespace { ... }`

Reply:
fix: https://github.com/dragonflydb/dragonfly/pull/2164/commits/1aaf2716a3f3cef6ee5702836c5c13d82757b192